src/materialized/src/lib.rs

```diff
@@ -380,58 +387,77 @@ async fn serve_stash<S: mz_stash::Append + 'static>(
         });
     }
 
+    // TODO(benesch): replace both `TCPListenerStream`s below with
+    // `<type>_listener.incoming()` if that is
+    // restored when the `Stream` trait stabilizes.
+
     // Launch task to serve connections.
     //
     // The lifetime of this task is controlled by a trigger that activates on
     // drop. Draining marks the beginning of the server shutdown process and
     // indicates that new user connections (i.e., pgwire and HTTP connections)
     // should be rejected. Once all existing user connections have gracefully
     // terminated, this task exits.
-    let (drain_trigger, drain_tripwire) = oneshot::channel();
+    let (sql_drain_trigger, sql_drain_tripwire) = oneshot::channel();
     task::spawn(|| "pgwire_server", {
         let pgwire_server = mz_pgwire::Server::new(mz_pgwire::Config {
             tls: pgwire_tls,
             coord_client: coord_client.clone(),
             metrics_registry: &config.metrics_registry,
             frontegg: config.frontegg.clone(),
         });
-        let http_server = http::Server::new(http::Config {
-            tls: http_tls,
-            frontegg: config.frontegg,
-            coord_client,
-            allowed_origin: config.cors_allowed_origin,
-        });
-        let mut mux = Mux::new();
-        mux.add_handler(pgwire_server);
-        mux.add_handler(http_server);
+
         async move {
-            // TODO(benesch): replace with `listener.incoming()` if that is
-            // restored when the `Stream` trait stabilizes.
-            let mut incoming = TcpListenerStream::new(listener);
-            mux.serve(incoming.by_ref().take_until(drain_tripwire))
+            let mut incoming = TcpListenerStream::new(sql_listener);
+            pgwire_server
+                .serve(incoming.by_ref().take_until(sql_drain_tripwire))
+                .await;
+        }
+    });
```

Comment:
We need another one of these for the internal pgwire server. It should look identical except it always sets `tls` and `frontegg` to `None`.